File: docs-examples/example-java/src/test/java/io/micronaut/configuration/openapi/docs/SecuritySchemeTest.java
package io.micronaut.configuration.openapi.docs;

import io.swagger.v3.oas.models.security.SecurityScheme;
import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;

class SecuritySchemeTest {

    @Test
    void oauth2SecurityScheme() {
        var openApi = OpenApiSpec.load();

        var scheme = openApi.getComponents().getSecuritySchemes().get("openid");
        assertEquals(SecurityScheme.Type.OAUTH2, scheme.getType());
        var flow = scheme.getFlows().getAuthorizationCode();
        assertEquals("https://mycompany.okta.com/oauth2/default/v1/authorize", flow.getAuthorizationUrl());
        assertEquals("https://mycompany.okta.com/oauth2/default/v1/token", flow.getTokenUrl());
        assertEquals("OpenID role", flow.getScopes().get("openid"));
    }

    @Test
    void securityRequirementOnTheController() {
        var operation = OpenApiSpec.load().getPaths().get("/").getGet();

        assertEquals(List.of("openid"), operation.getSecurity().get(0).get("openid"));
    }
}

File: docs-examples/example-python/src/test/python/micronaut/configuration/openapi/docs/SecuritySchemeTest.py
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test

from . import OpenApiSpec

from io.swagger.v3.oas.models.security.SecurityScheme import Type as SecuritySchemeType


@MicronautTest
class SecuritySchemeTest:

    @Test
    def test_oauth2_security_scheme(self):
        open_api = OpenApiSpec.load()

        scheme = open_api.getComponents().getSecuritySchemes().get("openid")
        assert scheme.getType() == SecuritySchemeType.OAUTH2
        flow = scheme.getFlows().getAuthorizationCode()
        assert flow.getAuthorizationUrl() == "https://mycompany.okta.com/oauth2/default/v1/authorize"
        assert flow.getTokenUrl() == "https://mycompany.okta.com/oauth2/default/v1/token"
        assert flow.getScopes().get("openid") == "OpenID role"

    @Test
    def test_security_requirement_on_the_controller(self):
        operation = OpenApiSpec.load().getPaths().get("/").getGet()

        assert list(operation.getSecurity().get(0).get("openid")) == ["openid"]
